server/service/orbit_client.go

```diff
@@ -37,29 +54,71 @@ func (oc *OrbitClient) request(verb string, path string, params interface{}, res
 	oc.setClientCapabilitiesHeader(request)
 	response, err := oc.http.Do(request)
 	if err != nil {
+		oc.setLastRequestError(err)
 		return fmt.Errorf("%s %s: %w", verb, path, err)
 	}
 	defer response.Body.Close()
 
-	return oc.parseResponse(verb, path, response, resp)
+	if err := oc.parseResponse(verb, path, response, resp); err != nil {
+		oc.setLastRequestError(err)
+		return err
+	}
+	return nil
```

Comment:
if we don't clear the value of `lastReqErr` after a request without errors (which makes sense) should we call it `lastRecordedErr` or similar?

Reply:
Good idea.